set.mm

```diff
@@ -24853,6 +24861,19 @@ replaced with setvar variables (see ~ cleljust ), so we don't include
       ( wceq wb eqeqan12d ancoms ) ABCEIDFIJABCDEFGHKL $.
   $}
 
+  ${
+    eqeqan1d.1 $e |- ( ph -> A = B ) $.
+    $( Implication of introducing a new equality.  (Contributed by Peter Mazsa,
+       17-Apr-2019.)  (Proof shortened by AV, 10-Feb-2023.) $)
+    eqeqan1d $p |- ( ( ph /\ C = D ) -> ( A = C <-> B = D ) ) $=
+      ( wceq id eqeqan12d ) ADEGZBCDEFJHI $.
```

Comment:
It has only two usages, so probably does not pay for itself.
eqeqan12d looks sufficient. Actually, what could be useful is an version of eqeqan12d with the same antecedent.

Reply:
see my comment above

Reply:
There are 5 (additional) proofs which could be shortened using ~eqeqan1d: ~chcoeffeq, ~gaorb, ~isdmn, ~isrgr, ~sylow1.
Is this a sufficient argument to move/keep ~eqeqan1d in main?

Reply:
Maybe. I will not oppose the move to Main, but I would like someone else (@tirix ?) to give their insight. Also, I think the label is not consistent with other theorems `eqeqan*` but you and Thierry know these labeling conventions better, so please just check that it's ok.

Reply:
I kind of agree with @benjub 's position [above](https://github.com/metamath/set.mm/pull/3034/files#r1103772266), which is especially valid in the case above.

Norm was saying that one could build a huge number of new first order logic theorems using two-steps proofs from the existing ones, but that he considered first order theory in `set.mm` as complete, and was reluctant to introduce such new theorems (I did not make the effort to find the exact reference and his exact words, but I clearly remember that). 

I think this would have applied here, and for `~orcomdd`.

As for naming, I think this is a kind of hybrid case, maybe a new naming convention would be useful?

Reply:
I think the label is OK, because the "2" in ~eqeqan12d is eliminated by merging the 2. hypothesis into the 2. part of the conjunction.

Reply:
> Norm was saying that one could build a huge number of new first order logic theorems using two-steps proofs from the existing ones, but [probably we shouldn't]

When you say it that way it sounds persuasive. I wish I had a better feel for the best place to draw the line. We can always fall back on rules like whether it decreases the total size of set.mm which is perhaps not a bad rule (even though it typically takes some effort to measure it in practice).

Reply:
Same as for ~orcomdd, I will mark this theorem as deprecated.

Reply:
Good for me.